applicationset/webhook/webhook.go

```diff
@@ -412,10 +412,12 @@ func shouldRefreshPRGenerator(gen *v1alpha1.PullRequestGenerator, info *prGenera
 	}
 
 	if gen.Github != nil && info.Github != nil {
-		if gen.Github.Owner != info.Github.Owner {
+		// repository owner and name are case-insensitive
+		// See https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests
+		if !strings.EqualFold(gen.Github.Owner, info.Github.Owner) {
 			return false
 		}
-		if gen.Github.Repo != info.Github.Repo {
+		if !strings.EqualFold(gen.Github.Repo, info.Github.Repo) {
```

Comment:
Wanted to point out that [EqualFold is not equivalent to comparing ToLower](https://github.com/golang/go/issues/52204), I am not sure if github owner/repo values can fall into this edge case but we should confirm before merging this change.

Reply:
@gdsoumya I'm pretty sure we won't be caught out by this as GitHub only allows ASCII names, so the more complex Unicode folding rules don't get involved.


For example, when using `µμ` as the name, it gets changed to a dash:

![image](https://github.com/argoproj/argo-cd/assets/2552620/f7355b23-0a9c-436d-8a42-337b98ca14ed)

This seems to be the same for Organisation names at least, and I presume it applies to users as well.